### PR TITLE
Disallow paths beginning with @ in templates

### PR DESCRIPTION
**Pull request by courajs**

### CODE REVIEW

**Comment by courajs** [comment]
ping @chancancode

**Comment by rwjblue** [comment]
This looks good to me, thank you @courajs!

**Comment by chadhietala** [comment]
Thanks @courajs!

File: packages/ember-template-compiler/tests/plugins/assert-reserved-named-arguments-test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    compile('{{#if @foo}}Yup{{/if}}', {
+      moduleName: 'baz/foo-bar'
+    });
+  }, `'@foo' is not a valid path. ('baz/foo-bar' @ L1:C6) `);
+
+  expectAssertion(() => {
+    compile('{{input type=(if @foo "bar" "baz")}}', {
+      moduleName: 'baz/foo-bar'
+    });
+  }, `'@foo' is not a valid path. ('baz/foo-bar' @ L1:C17) `);
```
Comment (chancancode) [inline packages/ember-template-compiler/tests/plugins/assert-reserved-named-arguments-test.js]: @courajs can you add another test for `<foo-bar @foo="bar" />`?

Comment (courajs) [inline packages/ember-template-compiler/tests/plugins/assert-reserved-named-arguments-test.js]: Sure, I can open another PR tmw. Any others I should include?

Comment (chancancode) [inline packages/ember-template-compiler/tests/plugins/assert-reserved-named-arguments-test.js]: Those are the main ones I can think of, thanks!

**Pull request merged**